v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/CassandraDriverConfigLoader.java

```diff
@@ -0,0 +1,147 @@
+import com.datastax.oss.driver.api.core.config.TypedDriverOption;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.ConfigException;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.util.Map.Entry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A common static utility class that allows the spanner migration pipelines to ingest Cassandra
+ * Driver config file from GCS. Cassandra has a structured config file to accept all the driver
+ * parameters, be it list of host ip addresses, credentials, retry policy and many more. Most of
+ * these parameters are very specific to the Cassandra Database. Refer to the <a
+ * href=>https://docs.datastax.com/en/developer/java-driver/4.3/manual/core/configuration/reference/index.html>reference
+ * configuration</a> for the file format.
+ */
+public final class CassandraDriverConfigLoader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CassandraDriverConfigLoader.class);
+  private static final ImmutableMap<String, TypedDriverOption> OPTIONS_SUPPORTED_BY_DRIVER =
+      getOptionsSupportedByDriver();
+
+  /**
+   * Load the Cassandra Config from a file as a {@link DriverConfigLoader}.
+   *
+   * @param path A complete gcs path to the config file of the form "gs://path/to/file".
+   * @return DriverConfigLoader.
+   * @throws FileNotFoundException - If file is not found at specified path.
+   */
+  public static DriverConfigLoader load(String path) throws FileNotFoundException {
+    URL url = loadSingleFile(path);
+    LOG.debug("Loaded Cassandra Driver config from path {}", path);
+    try {
+      DriverConfigLoader.fromUrl(url).getInitialConfig();
+      return DriverConfigLoader.fromUrl(url);
+    } catch (ConfigException.Parse parseException) {
+      LOG.error(
+          "Parsing error while parsing Cassandra Driver config from path {}", path, parseException);
+      throw parseException;
+    }
+  }
+
+  /**
+   * Load the Cassandra Config from a file as a {@link java.io.Serializable} {@link OptionsMap}.
+   * This {@link OptionsMap} can be stored in any object that needs to implement {@link
+   * java.io.Serializable}. At the time of opening a conneciton to Cassandra, it can be deserialized
+   * by {@link CassandraDriverConfigLoader#fromOptionsMap(OptionsMap)}. Note: Implementation Detail,
+   * Cassandra Driver does not provide a direct method to convert a link {@link DriverConfigLoader}
+   * into an {@link OptionsMap}, or build an {@link OptionsMap} from a file.
+   *
+   * @param path A complete gcs path to the config file of the form "gs://path/to/file".
+   * @return DriverConfigLoader.
+   * @throws FileNotFoundException - If file is not found at specified path.
+   */
+  public static OptionsMap loadToSerializableOptionsMap(String path) throws FileNotFoundException {
```

Comment:
nit: rename to be clearer - something like getOptionsMapFromFile or something.

Reply:
Thanks. Fixed in current push.